catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CreateOperations.java

```diff
@@ -0,0 +1,589 @@
+import ddf.catalog.content.operation.StorageRequest;
+import ddf.catalog.content.operation.impl.CreateStorageRequestImpl;
+import ddf.catalog.content.plugin.PostCreateStoragePlugin;
+import ddf.catalog.content.plugin.PreCreateStoragePlugin;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.history.Historian;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.OperationTransaction;
+import ddf.catalog.operation.ProcessingDetails;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.CreateResponseImpl;
+import ddf.catalog.operation.impl.OperationTransactionImpl;
+import ddf.catalog.operation.impl.ProcessingDetailsImpl;
+import ddf.catalog.plugin.AccessPlugin;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.PostIngestPlugin;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.CatalogStore;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.InternalIngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.util.impl.Requests;
+
+public class CreateOperations {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateOperations.class);
+
+    private static final Logger INGEST_LOGGER =
+            LoggerFactory.getLogger(Constants.INGEST_LOGGER_NAME);
+
+    private static final String PRE_INGEST_ERROR =
+            "Error during pre-ingest service invocation:\n\n";
+
+    private Supplier<CatalogProvider> catalogSupplier;
+
+    private Supplier<StorageProvider> storageSupplier;
+
+    // Inject properties
+    private FrameworkProperties frameworkProperties;
+
+    private Historian historian;
+
+    private QueryOperations queryOperations;
+
+    private SourceOperations sourceOperations;
+
+    private OperationsSecuritySupport opsSecuritySupport;
+
+    private OperationsMetacardSupport opsMetacardSupport;
+
+    private OperationsCrudSupport opsCrudSupport;
+
+    public CreateOperations(FrameworkProperties frameworkProperties) {
+        this.frameworkProperties = frameworkProperties;
+    }
+
+    void setCatalogSupplier(Supplier<CatalogProvider> catalogSupplier) {
+        this.catalogSupplier = catalogSupplier;
+    }
+
+    void setStorageSupplier(Supplier<StorageProvider> storageSupplier) {
+        this.storageSupplier = storageSupplier;
+    }
+
+    public void setHistorian(Historian historian) {
+        this.historian = historian;
+    }
+
+    public void setQueryOperations(QueryOperations queryOperations) {
+        this.queryOperations = queryOperations;
+    }
+
+    public void setSourceOperations(SourceOperations sourceOperations) {
```

Comment:
That's a good point. I think I can move the operations and ops-support injections to be through ctor throughout.